pg_query.h

```diff
@@ -119,6 +125,8 @@ PgQuerySplitResult pg_query_split_with_parser(const char *input);
 
 PgQueryDeparseResult pg_query_deparse_protobuf(PgQueryProtobuf parse_tree);
 
+PgQuerySummaryParseResult pg_query_parse_summary(const char* input, int parser_options, int truncate_limit);
```

Comment:
We might want to add a `_protobuf` suffix on the function name to follow the existing naming convention. Thoughts @lfittl?

Also `parse_summary` might be too verbose. Could it just be `summary`?

Then we'd have `pg_query_summary_protobuf`

Reply:
I was wondering about removing the `parse_` part of the name as well. I think `pg_query_summary_protobuf` would be fine. :+1: